### PR TITLE
feat: add UserAgentManager

### PR DESCRIPTION
**Pull request by breedloj**
## Description
Adds a static UserAgentManager which manages environment level SDK user agent information. User agent component registration and full user agent string retrieval is encapsulated by this new class rather than being registered and computed at the CustomSkill layer.

The manager does not allow for duplicate user agent components and only computes an updated user agent string once per valid component addition. At request time, the precomputed value is used rather than the existing approach which required reconstruction per request.

## Motivation and Context
The primary motivation for this change is to allow other frameworks that are designed for use with the SDK, but may not directly be configured on the SDK instance itself, to register themselves with user agent components. With this singleton manager approach, such components can directly append a user agent component at runtime/request execution.

## Testing
Added tests for the UserAgentManager and an additional test for CustomSkill. Existing tests continue to pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: ask-sdk-core/tst/skill/CustomSkill.spec.ts [hunk 1]
```diff
@@ -20,6 +21,10 @@ import { JsonProvider } from '../mocks/JsonProvider';
 import { MockAlwaysFalseRequestHandler } from '../mocks/request/MockAlwaysFalseRequestHandler';
 import { MockAlwaysTrueRequestHandler } from '../mocks/request/MockAlwaysTrueRequestHandler';
 
+beforeEach(function () {
```
Comment (ShenChen93) [inline ask-sdk-core/tst/skill/CustomSkill.spec.ts]: lint: unexpected space before function parentheses

File: ask-sdk-runtime/lib/util/UserAgentManager.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    /**
+     * Retrieves the full user agent string, containing all registered components.
+     */
+    static getUserAgent(): string {
+        return this.userAgent;
+    }
+
+    /**
+     * Registers a user agent component. This will be appeneded to the generated
```
Comment (ShenChen93) [inline ask-sdk-runtime/lib/util/UserAgentManager.ts]: nit: appended

**Review by nikhilym — APPROVED** [review]
LGTM. Looks like the tests do not install the source from github but instead from npm, thus leading to the failure. Pulled in locally and could get the tests working fine. 🚢

**Review by nikhilym — APPROVED** [review]

File: ask-sdk-runtime/lib/util/UserAgentManager.ts (hunk 2 above)
Comment (breedloj) [reply ask-sdk-runtime/lib/util/UserAgentManager.ts]: Good catch! Fixed.

File: ask-sdk-core/tst/skill/CustomSkill.spec.ts (hunk 1 above)
Comment (breedloj) [reply ask-sdk-core/tst/skill/CustomSkill.spec.ts]: Already fixed :D

**Review by ShenChen93 — APPROVED** [review]

**Pull request merged**